CallAutomation_OutboundCalling/CallAutomation_OutboundCalling/Program.cs

```diff
@@ -0,0 +1,108 @@
+
+// Base url of the app
+var callbackUriHost = Environment.GetEnvironmentVariable("VS_TUNNEL_URL")?.TrimEnd('/');
+
+CallAutomationClient callAutomationClient = new CallAutomationClient(acsConnectionString);
+builder.Services.AddSingleton(callAutomationClient);
+var app = builder.Build();
+
+app.MapPost("/api/outboundCall", async (ILogger<Program> logger) =>
+{
+    PhoneNumberIdentifier target = new PhoneNumberIdentifier(targetPhonenumber);
+    PhoneNumberIdentifier caller = new PhoneNumberIdentifier(acsPhonenumber);
+    CallInvite callInvite = new CallInvite(target, caller);
+    CreateCallResult createCallResult = await callAutomationClient.CreateCallAsync(callInvite, new Uri(callbackUriHost + "/api/callbacks")).ConfigureAwait(false);
+
+    logger.LogInformation($"Reponse from CallConnection Id : {createCallResult.CallConnectionProperties.CallConnectionId}");
+});
+
+app.MapPost("/api/callbacks", async (CloudEvent[] cloudEvents, ILogger<Program> logger) =>
+{
+    foreach (var cloudEvent in cloudEvents)
+    {
+        CallAutomationEventBase @event = CallAutomationEventParser.Parse(cloudEvent);
+        logger.LogInformation($"Event Recieved. CallConnectionId[{@event?.CallConnectionId}], Type Name[{@event?.GetType().Name}]");
+        var callConnection = callAutomationClient.GetCallConnection(@event.CallConnectionId);
+        var callMedia = callConnection.GetCallMedia();
+
+        if (@event is CallConnected)
+        {
+            logger.LogInformation($"Start Recording...");
+            CallLocator callLocator = new ServerCallLocator(@event.ServerCallId);
+            _ = await callAutomationClient.GetCallRecording().StartAsync(new StartRecordingOptions(callLocator));
+
+            // prepare recognize tones
+            CallMediaRecognizeDtmfOptions callMediaRecognizeDtmfOptions = new CallMediaRecognizeDtmfOptions(new PhoneNumberIdentifier(targetPhonenumber), maxTonesToCollect: 1);
+            callMediaRecognizeDtmfOptions.Prompt = new FileSource(new Uri(callbackUriHost + "/audio/MainMenu.wav"));
+            callMediaRecognizeDtmfOptions.InterruptPrompt = true;
+            callMediaRecognizeDtmfOptions.InitialSilenceTimeout = TimeSpan.FromSeconds(5);
+            callMediaRecognizeDtmfOptions.InterToneTimeout = TimeSpan.FromSeconds(10);
+            callMediaRecognizeDtmfOptions.StopTones = new List<DtmfTone> { DtmfTone.Pound, DtmfTone.Asterisk };
+
+            // Send request to recognize tones
+            await callMedia.StartRecognizingAsync(callMediaRecognizeDtmfOptions);
+        }
+        if (@event is RecognizeCompleted)
+        {
+            // Play audio once recognition is completed sucessfully
+            logger.LogInformation($"RecognizeCompleted event received for call connection id: {@event.CallConnectionId}");
+            var recognizeCompleted = (RecognizeCompleted)@event;
+
+            string selectedTone = ((DtmfResult)recognizeCompleted.RecognizeResult).ConvertToString();
```

Comment:
Could remove the log $"RecognizeCompleted event received for call connection id: {@event.CallConnectionId}" since it's redundant with the above log and instead log the value of selectedTone:
logger.LogInformation($"Received DTMF tone {selectedTone}");